utilities/infra.py

```diff
@@ -312,20 +312,29 @@ def get_not_running_pods(pods: list[Pod], filter_pods_by_name: str = "") -> list
     return pods_not_running
 
 
-def wait_for_pods_running(
+def wait_for_pods_running(  # type: ignore[return]
     admin_client: DynamicClient,
     namespace: Namespace,
     number_of_consecutive_checks: int = 1,
     filter_pods_by_name: str = "",
-) -> None:
+    raise_exception: bool = True,
```

Comment:
I would recommend to avoid passing such bools as arguments.
Any boolean that is conditioned on in the function, is a smell. Usually it means you need another function and one can wrap the other per need (or extract to a common function the relevant parts).

I is also a smell when you need to differentiate between a `None` and a `list` on the return value.
When you have these two options, it means the caller needs to check it for `None` all the time.

Reply:
i need to diff between raise and return to allow accumulating the failures (e.g network sanity)